### PR TITLE
Add test for storage iteration with broken values

### PR DESCRIPTION
**Pull request by SupunS**

### CODE REVIEW

**Review by turbolent — APPROVED** [review]
Nice

File: fvm/fvm_test.go [hunk 1]
```diff
@@ -2448,3 +2449,145 @@ func TestAttachments(t *testing.T) {
...
+		).
+		run(
+			func(
+				t *testing.T,
+				vm fvm.VM,
+				chain flow.Chain,
+				ctx fvm.Context,
+				snapshotTree snapshot.SnapshotTree,
+			) {
+				// Create two private keys
```
Comment (janezpodhostnik) [inline fvm/fvm_test.go:2478]: incorrect comment.

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/fvm_test.go (hunk 1 above)
Comment (turbolent) [inline fvm/fvm_test.go:2478]:
```suggestion
				// Create a private key
```

File: fvm/fvm_test.go [hunk 2]
```diff
@@ -2448,3 +2449,145 @@ func TestAttachments(t *testing.T) {
...
+				privateKeys, err := testutil.GenerateAccountPrivateKeys(1)
+				require.NoError(t, err)
+
+				// Bootstrap a ledger, creating accounts with the provided private keys and the root account.
```
Comment (turbolent) [inline fvm/fvm_test.go:2482]:
```suggestion
				// Bootstrap a ledger, creating an account with the provided private key and the root account.
```

**Comment by turbolent** [comment]
bors merge

**Comment by SupunS** [comment]
bors merge

**Comment by turbolent** [comment]
bors retry

**Pull request merged**